### PR TITLE
[18.09] Add Loom datatype and remove rdata.seurat

### PR DESCRIPTION
**Pull request by blankenberg**

### CODE REVIEW

**Comment by blankenberg** [comment]
@galaxybot test this

**Comment by blankenberg** [comment]
@galaxybot test this

**Comment by jxtx** [comment]
image

**Review by bgruening — APPROVED** [review]
Thanks @blankenberg! Looks good to me.
@mblue9 any objections to remove rdata.seurat in favor of loom for this release?

Not sure why tests are failing :(

**Comment by mblue9** [comment]
Fine by me 👍

**Comment by dannon** [comment]
Rerunning the integration tests to see what's up.

**Comment by mvdbeek** [comment]
The previous integration passed but didn't shut down correctly. The error now is unrelated.

**Comment by dannon** [comment]
@mvdbeek Yeah, I dug through, too, and it looked unrelated to me.  Queued it for another run just to see what happens with that test, though

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -736,6 +736,97 @@ def display_peek(self, dataset):
...
+                dataset.metadata.description = loom_file.attrs.get('description', None)
+                dataset.metadata.url = loom_file.attrs.get('url', None)
+                dataset.metadata.doi = loom_file.attrs.get('doi', None)
+                dataset.metadata.loom_spec_version = loom_file.attrs.get('LOOM_SPEC_VERSION', None)
+                dataset.creation_date = loom_file.attrs.get('creation_date', None)
+                dataset.metadata.shape = tuple(loom_file['matrix'].shape)
+
+                tmp = list(loom_file['layers'].keys())
+                dataset.metadata.layers_count = len(tmp)
+                dataset.metadata.layers_names = tmp
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/binary.py:809]: If loom_file['layers'] and friends aren't some kind of OrderedDict we should probably sort here

Comment (blankenberg) [reply lib/galaxy/datatypes/binary.py:809]: Its not technically a dictionary at all. Its an object that implements several bits of the dictionary style interface. My understanding is that h5py returns things in the same order as the standard tools (alphanumerically sorted), which is probably the desired behavior here, but I haven't done any extensive testing here. Presumably, if a specific tool needs a different ordering, then a sort filter could be applied within the parameter dropdown. Although if we really think we need to manually resort things, we can.

Comment (mvdbeek) [reply lib/galaxy/datatypes/binary.py:809]: I'm thinking about tool tests and equality of metadata when caching jobs here. If some of the metadata elements are returned with a random order that might be a little annoying. If it's the same order it's fine for me, and we can always fix it later (if it's not).

**Review by mvdbeek — APPROVED** [review]

**Comment by dannon** [comment]
The data manager test that failed before passed this time, too!  We'll have to track that down some other time.

**Comment by blankenberg** [comment]
Off topic, but worth burning an [hour](https://www.youtube.com/watch?v=z1aVDael-KM) of time...

**Comment by blankenberg** [comment]
Thanks all for the comments, reviews, and getting the tests to finally pass, I had started to give up after the 3rd-or-so unrelated failure.

**Pull request merged**